src/com/facebook/buck/maven/ArtifactConfig.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2015-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.maven;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+
+import org.kohsuke.args4j.Argument;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.kohsuke.args4j.Option;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ArtifactConfig {
+
+  @VisibleForTesting
+  static class CmdLineArgs {
+    @Argument(usage = "One or more artifacts to resolve", metaVar = "artifact")
+    public List<String> artifacts = new ArrayList<>();
+
+    @Option(name = "-repo", usage = "Root of your repository")
+    public String buckRepoRoot;
+
+    @Option(name = "-third-party", usage = "Directory to place dependencies in")
+    public String thirdParty = "third-party";
+
+    @Option(name = "-local-maven", usage = "Local Maven repository")
+    public String mavenLocalRepo;
+
+    @Option(name = "-maven", usage = "Maven URI(s)")
+    public List<String> repositoryURIs = new ArrayList<>();
+
+    @Option(name = "-visibility", usage = "Targets that can see the artifacts")
+    public List<String> visibility = new ArrayList<>();
+
+    @Option(name = "-json", usage = "JSON configuration file for artifacts, paths and maven repos")
+    public String artifactConfigJson;
+
+    @Option(name = "-help", help = true)
+    public boolean showHelp;
```

Comment:
That's what args4j seems to [suggest](https://github.com/kohsuke/args4j/blob/master/args4j/examples/SampleMain.java).